server/src/main/java/org/elasticsearch/index/codec/PerFieldMapperCodec.java

```diff
@@ -84,6 +93,30 @@ public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
 
     @Override
     public DocValuesFormat getDocValuesFormatForField(String field) {
+        if (useTSDBDocValuesFormat(field)) {
+            return tsdbDocValuesFormat;
+        }
         return docValuesFormat;
     }
+
+    private boolean useTSDBDocValuesFormat(final String field) {
+        return isTimeSeriesModeIndex() && isES97OrAbove() && (isTimestampField(field) || isTimeSeriesCounter(field));
+    }
+
+    private boolean isES97OrAbove() {
+        return indexSettings.getIndexMetadata().getCreationVersion().onOrAfter(Version.V_8_7_0);
+    }
+
+    private boolean isTimeSeriesCounter(String field) {
+        return mapperService.fieldType(field) != null
+            && TimeSeriesParams.MetricType.counter.equals(mapperService.fieldType(field).getMetricType());
+    }
+
+    private boolean isTimestampField(String field) {
+        return mapperService.mappingLookup().hasTimestampField() && DataStream.TimestampField.FIXED_TIMESTAMP_FIELD.equals(field);
+    }
```

Comment:
Maybe we should use it for all numeric fields? Gauge fields could benefit from it too if they have runs of equal values. And I could imagine us later on improve this doc-values format to handle the case of double values that change slowly over time.